pandas/core/apply.py

```diff
@@ -368,8 +377,11 @@ def agg_dict_like(self) -> DataFrame | Series:
         from pandas import Index
         from pandas.core.reshape.concat import concat
 
+        filtered_col = self._filter_numeric_only()
+
         obj = self.obj
         arg = cast(AggFuncTypeDict, self.f)
+        arg = {k: arg[k] for k in arg.keys() if k not in filtered_col}
```

Comment:
I specified modifying agg_dict_like before, but I'm now thinking I'm that we shouldn't be.  E.g. `b` a non-numeric-column:

    df.agg({"a": "sum", "b": "mean"}, numeric_only=True)

I don't think there is any expectation that you can pass `numeric_only` here. Apologies for sending you down the wrong path - unless I'm missing something.